snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -253,6 +254,11 @@ def _inject_snapcraft(self, *, new_container: bool):
         if common.is_snap():
             with tempfile.TemporaryDirectory(
                     prefix='snapcraft', dir=self._lxd_common_dir) as tmp_dir:
+                # Wait for any on-going refreshes to finish.
+                # If there are no changes an error will be returned.
+                with contextlib.suppress(ContainerRunError):
+                    self._container_run([
+                        'snap', 'watch', '--last=auto-refresh'])
```

Comment:
Will this print anything if it does actually wait?

Reply:
It will trigger the same progress bar that other commands show.